apps/teamsjs-cc-app/tabs/src/components/Tab.tsx

```diff
@@ -0,0 +1,330 @@
+        ],
+      },
+      {
+        key: 20,
+        items: [
+          { key: '20-1', content: <><Fluent.CallVideoIcon />Video</>, value: 'Video' },
+          { key: '20-2', content: VideoIsSupported() },
+          { key: '20-3', content: <Video />, className: 'ui_action' }
+        ],
+      },
+      {
+        key: 21,
+        items: [
+          { key: '21-1', content: <><Fluent.BriefcaseIcon />Web Storage</>, value: 'Web Storage' },
+          { key: '21-2', content: WebStorageIsSupported() },
+          { key: '21-3', content: <WebStorage />, className: 'ui_action' }
+        ],
+      }
+    ];
+  }
+
+  const updateCapabilityOnchange = (text: string) => {
+    setData().then((defaultRows) => {
+      if (showSupportedOnly) setShowSupportedOnly(false);
+
+      const rows = defaultRows.filter((defaultRow) => {
+        if (defaultRow.items[0].value?.toLowerCase()?.search(text.toLowerCase()) !== -1) {
+          return defaultRow;
+        }
+      });
+      setTableRows(rows);
+    }, (error) => {
+      console.log("Error", error);
+    })
+  }
+
+  useEffect(() => {
+    setData().then((defaultRows) => {
+      if (showSupportedOnly) {
+        const rows = defaultRows.filter((r) => { return r.items[1].content === 'Yes' });
+        setTableRows(rows);
+      } else {
+        setTableRows(defaultRows);
+      }
+    }, (error) => {
+      console.log("Error", error);
+    })
+
+  }, [showSupportedOnly]);
+
+  return (
+    <div className={themeString === "default" ? "" : "dark"}>
+      <Fluent.Flex column={true} gap={"gap.small"} padding={"padding.medium"} >
+        <Fluent.Segment>
+          <Hub />
+        </Fluent.Segment>
+        <Fluent.Segment>
+          <Fluent.Flex space="between">
+            <Fluent.Checkbox
+              label="Show supported only"
+              checked={showSupportedOnly}
+              onClick={() => setShowSupportedOnly(!showSupportedOnly)}
+              toggle />
+            <Fluent.Flex gap="gap.small">
+              <Fluent.Label>{packageJSON.dependencies["@microsoft/teams-js"]}</Fluent.Label>
+            </Fluent.Flex>
+          </Fluent.Flex>
+        </Fluent.Segment>
+        <Fluent.Segment>
+          <Fluent.Flex gap="gap.small" styles={{ justifyContent: 'space-between' }}>
+            <Fluent.Flex.Item>
+              <Fluent.Input icon={<Fluent.SearchIcon />} placeholder="Search capability" onChange={(e: any) => {
+                const event = e as React.SyntheticEvent<HTMLInputElement, Event>;
+                updateCapabilityOnchange(event.currentTarget.value);
+              }} />
+            </Fluent.Flex.Item>
+            <Fluent.Flex.Item>
+              <Fluent.Button onClick={() => {
+                setData().then((defaultRows) => {
+                  const defaultRowsString = JSON.stringify(defaultRows.map(x => {
+                    const arr1 = x.items.map((y, i) => {
+                      if (i === 2) return undefined;
```

Comment:
What you're doing here will totally work, but it's kind of an unusual way to accomplish this. Normally, I'd expect setData to return a collection of objects that all implement some sort of interface. Something like:

```
interface ICapability {
  icon: React.Element;
  name: string;
  isSupported: boolean;
  content: React.Element;
  className: string;
}
```

Then code like this could iterate over defaultRows and have code that looked like this:

```
return { Capability: x.name, Supported: x.isSupported }
```

Reply:
I think that might also remove some of your need to use map so often